velox/functions/prestosql/types/TimestampWithTimeZoneType.h

```diff
@@ -125,4 +124,18 @@ class TimestampWithTimeZoneTypeFactories : public CustomTypeFactories {
 
 void registerTimestampWithTimeZoneType();
 
+static constexpr int32_t kTimezoneMask = 0xFFF;
+static constexpr int32_t kMillisShift = 12;
+
+inline int64_t unpackMillisUtc(int64_t dateTimeWithTimeZone) {
```

Comment:
Looking at the usage, it looks like we always convert the return value to `Timestamp` using `Timestamp::fromMillis`. Does it make sense to return Timestamp value here instead? That will make the API cleaner.

Reply:
Maybe say `unpackTimestampUtc`.

Reply:
These methods will also be used by the `TimeWithTimezone` type in the future, so return Timestamp might not be appropriate.

Reply:
Should we have both `unpackMillisUtc` and `unpackTimestampUtc`?

Reply:
Refactored. Thank you @majetideepak deepak.